src/schema/v1/filter_artworks.ts

```diff
@@ -384,13 +359,70 @@ const filterArtworksTypeFactory = (
       loader = loaderWithCache
     }
 
-    const blacklistedFields = ["artworks_connection", "__id"]
-    if (queriedForFieldsOtherThanBlacklisted(fieldNodes, blacklistedFields)) {
-      return loader(gravityOptions).then(response =>
-        Object.assign({}, response, { options: gravityOptions })
-      )
+    // Introspect connection args for `artworks_connection`.
+    // Use those and fetch all data in one request.
+    // Resolver for `artworks_connection` just creates a connection
+    // from the already fetched data, instead of re-fetching.
+
+    const connectionArgs = {}
+    info.fieldNodes &&
```

Comment:
this `resolver` is now really long with this added extra logic, would it make sense to make this visitor logic it's own method and call it from here?

Reply:
I concur, also because I consider `SELECTION_DEPTH_THRESHOLD` being private to the module preferable.

Reply:
Will move this out!